aquadoggo/src/schema_service/temp_file.rs

```diff
@@ -0,0 +1,77 @@
+// SPDX-License-Identifier: AGPL-3.0-or-later
+
+use std::fs::File;
+use std::io::BufReader;
+use std::marker::PhantomData;
+
+use serde::de::DeserializeOwned;
+use serde::Serialize;
+
+/// This helper allows circumventing visibility limitations by storing/loading data in a temp file.
+pub(crate) struct TempFile<Data>
+where
+    Data: Serialize + DeserializeOwned,
+{
+    /// Path of the temporary file.
+    path: String,
+
+    // Marker for generic type of stored data.
+    data_type: PhantomData<Data>,
+}
+
+impl<Data: Serialize + DeserializeOwned> TempFile<Data> {
+    /// Restore data given the temporary files path.
+    pub fn load(path: &str) -> Data {
+        let file = File::open(path).unwrap();
+        let reader = BufReader::new(file);
+        serde_json::from_reader(reader).unwrap()
+    }
+
+    /// Restore data and pin it to memory for the rest of the program's runtime.
+    ///
+    /// **Attention**: dropping the returned reference will leak memory.
+    pub fn load_static(path: &str) -> &'static Data {
+        Box::leak(Box::new(Self::load(path)))
```

Comment:
I wonder if we can make this whole "file thing" a static object living throughout the program by using `lazy_static` or the more fresh [`once_cell`](https://docs.rs/once_cell/latest/once_cell/index.html)